lib/partitioner/go.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'cocaine'
+require 'set'
+require 'partitioner/dependency_analysis'
+
+module Partitioner
+  # This partitioner shard Go repos
+  class Go < DependencyAnalysis
+
+    def initialize(build, kochiku_yml)
+      super(build, kochiku_yml, {})
+    end
+
+    def partitioner_type
+      'go'
+    end
+
+    def all_modules
+      return @all_modules if @all_modules
+
+      @all_moduless = module_dependency_map.keys
+    end
+
+    def module_dependency_map
+      return @module_dependency_map if @module_dependency_map
+
+      @module_dependency_map = {}
+
+      # NOTE(kaneda): THIS MIGHT NOT WORK?
+      GitRepo.inside_copy(@build.repository, @build.ref) do
+        outputs = Cocaine::CommandLine.new("go list -f '{{.ImportPath}}:{{.Imports}}' ./...").run.split("\n")
```

Comment:
I don't know if it helps, but this output also computes the transitive dependencies for the given packages. Look in the `.Deps` field.

Reply:
Yeah the existing Kochiku code has code to calculate transitive dependencies. We can reuse it.